client/src/util/college-form-schema.ts

```diff
@@ -0,0 +1,37 @@
+import {z} from "zod";
+import {CategoryField, CollegeFormField} from "./types/college-form";
+import {collegeForm} from "./data/college-form";
+
+const createZodField = (field: CollegeFormField) => {
+  switch (field.type) {
+    case "text":
+    case "url":
+    case "date":
+    case "longform":
+      return z.string().refine((data) => !field.required || data.trim() !== '', {
+        message: "Field is required",
+      });
+    case "number":
+      return z.string().transform(v => parseInt(v, 10)).refine((data) => !field.required || !isNaN(data), {
+        message: "Field must be a number",
+      });
+    default:
+      return z.unknown(); // You can customize this for other field types
+  }
+};
+
+const processFields = (fields: (CategoryField | CollegeFormField)[]) => {
+  const zodFields: Record<string, ReturnType<typeof createZodField>> = {};
+
+  fields.forEach((field) => {
+    if (field.type === "category") {
+      processFields(field.fields);
+    } else {
+      zodFields[field.id] = createZodField(field)
+    }
+  });
+
+  return zodFields;
+};
```

Comment:
this logic is used to create schema validators for the college form sections. right now, it validates that "number" fields are numbers, and that required fields have text.

in the future, we could consider adding proper validation for URLs (dates are already natively validated in-browser), but since none of the URL fields are required i didn't add that yet.